core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -114,28 +113,21 @@ public QueryShardContext(QueryShardContext source) {
         this.types = source.getTypes();
     }
 
-    public void parseFieldMatcher(ParseFieldMatcher parseFieldMatcher) {
-        this.parseContext.parseFieldMatcher(parseFieldMatcher);
-    }
-
+    /**
+     * @return the parseFieldMatcher configured in the index settings
+     */
     public ParseFieldMatcher parseFieldMatcher() {
-        return parseContext.parseFieldMatcher();
+        return indexSettings.getParseFieldMatcher();
```

Comment:
++ to that. In the tests its good though that some methods take the matcher as an argument, so we can test thing in strict and non-strict mode without having pass IndexSettings around.